### ISSUE TITLE
Support for IOS

### ISSUE THREAD

**Issue by Lulu-Games** [SuperTux/supertux#2738]
I would really like to have Supertux on my IPhone. Is that possible or is there a problem with the license? (Supertuxkart is also available for IOS)

**Comment by tulpenkiste** [comment]
It is technically possible but it would require an active maintainer or two for it to even potentially happen (and no one in the SuperTux team actually wants to deal with that or the tedious process of actually signing iOS executables)

**Comment by Lulu-Games** [comment]
OK, but maybe only the Stable releases.

**Comment by Lulu-Games** [comment]
Or even make the web assembly Phone compatible

**Comment by tulpenkiste** [comment]
> OK, but maybe only the Stable releases.
Again, its still not worth doing for the SuperTux team, especially as making iOS executables signed (so it can be installed) is an expensive process, requiring a mac device (or hackintosh) and an expensive subscription (100 USD per year irrc) for access to the Apple Developer Program so you can merely start signing executables and distributing the game on the App Store. I really don't think thats feasible or something the SuperTux team would want to deal with, at least right now.

**Comment by Lulu-Games** [comment]
> > OK, but maybe only the Stable releases.
> 
> Again, its still not worth doing for the SuperTux team, especially as making iOS executables signed (so it can be installed) is an expensive process, requiring a mac device (or hackintosh) and an expensive subscription (100 USD per year irrc) for access to the Apple Developer Program so you can merely start signing executables and distributing the game on the App Store. I really don't think thats feasible or something the SuperTux team would want to deal with, at least right now.

I accept that! That is too much money for nothing.

**Comment by MatusGuy** [comment]
making wasm playable is probably doable

**Comment by Lulu-Games** [comment]
> making wasm playable is probably doable

I also think so, currently you get errors, if you try to run it on an IPhone. But you would also need a virtual joystick. Do you need to pay only for the App store or also for the TestFlight beta?

**Comment by tulpenkiste** [comment]
> > making wasm playable is probably doable
> 
> I also think so, currently you get errors, if you try to run it on an IPhone. But you would also need a virtual joystick. Do you need to pay only for the App store or also for the TestFlight beta?

Not sure about paying for the app store or testflight. I only know of the Developer Account costs.
In terms of virtual joystick, theres already one built in and it should auto activate, but you can manually turn it on if needed.

**Comment by Brockengespenst** [comment]
Technically interesting, legally probably not so interesting. As far as I know, the GPL and Apple's App store are not good friends (you will find a (pretty old) statement from the free software foundation regarding that topic). STK is not available in the App store, only as "beta" via Testflight - not sure if the conditions for Testflight differ from those in the App Store.
For just compiling and putting the result on your Apple device, there is no need for a developer account, just XCode and an apple device. The app can be used for one week (at least the last time I tried several years ago), after that you have to install it again using XCode and can use it again for one week and so on.
So just for developing, no need to pay. Only when it comes to distribution of the app via App Store (probably also via Testflight, but not sure about this), you need the developer account and pay.

**Comment by Anatoliy6463** [comment]
That will be cool!(I use android)

**Comment by tulpenkiste** [comment]
> just remember the time when VLC was removed from the App Store due to GPL licence problems. Looks like they have changed the licence to LGPL and nowadays it is back in the App Store

That is not true, VLC's license has not been updated ever. It's still dual licensed under GPL-2.0 and MPL-2.0 and has been that way for 11 years.
The only VLC thing that has had a license change is libvlc, which went to LGPL-2.1 but the actual iOS app is still GPL-2.0/MPL-2.0.

**Comment by Anatoliy6463** [comment]
OK, I will create a fork of SuperTux with BSD license instead of GPL to make it published in AppStore. Or you just can jailbreak your iPhone(I don't know is it legal in Germany).

**Comment by tulpenkiste** [comment]
> OK, I will create a fork of SuperTux with BSD license instead of GPL to make it published in AppStore

You're probably joking but if you're not, I need to specify that relicensing GPL software to anything else is illegal unless you have (written) consent from every copyright holder (in this case, anyone who has contributed to SuperTux ever).

> Or you just can jailbreak your iPhone(I don't know is it legal in Germany).

This on the other hand is entirely legal afaik.

**Comment by MatusGuy** [comment]
> OK, I will create a fork of SuperTux with BSD license instead of GPL to make it published in AppStore.

Chad

**Comment by Anatoliy6463** [comment]
> > OK, I will create a fork of SuperTux with BSD license instead of GPL to make it published in AppStore.
> 
> Chad

Oh, i forgot

**Comment by Lulu-Games** [comment]
OK fine but why doesn't WASM work in all my browsers (I have an IPhone 7). I get out of memory exceptions

**Comment by Anatoliy6463** [comment]
> OK fine but why doesn't WASM work in all my browsers (I have an IPhone 7). I get out of memory exceptions

Jailbreak your phone ;)

**Comment by Anatoliy6463** [comment]
> > OK fine but why doesn't WASM work in all my browsers (I have an IPhone 7). I get out of memory exceptions
> 
> Jailbreak your phone ;)

Or close background apps, because they use memory and CPU

**Comment by tulpenkiste** [comment]
> OK fine but why doesn't WASM work in all my browsers (I have an IPhone 7). I get out of memory exceptions

If I had to hazard a guess, your phone might not have enough RAM for SuperTux? I know that Android uses a lot of RAM in of itself and I'm imagining iOS 12 does the same thing. I did just make a PR (#2745) that adds a cap to the RAM but I doubt it'll fix the issue.

**Comment by Lulu-Games** [comment]
But I get the same problem on a IPAD pro (with defenetly enough ram)

**Comment by Lulu-Games** [comment]
> > OK fine but why doesn't WASM work in all my browsers (I have an IPhone 7). I get out of memory exceptions
> 
> Jailbreak your phone ;)

I'm happy with IOS because of its security. Wich OS should I install after a Jailbreak?

**Comment by Anatoliy6463** [comment]
> > > OK fine but why doesn't WASM work in all my browsers (I have an IPhone 7). I get out of memory exceptions
> > 
> > 
> > Jailbreak your phone ;)
> 
> I'm happy with IOS because of its security. Wich OS should I install after a Jailbreak? (I HATE android)

Jailbreak just removes Apple's DRM and you can install any apps. But if you want to use other OS, try Fedora Phosh, Mobian or Ubuntu Touch.

**Comment by Anatoliy6463** [comment]
> But I get the same problem on a IPAD pro (with defenetly enough ram)

I think it's because of old iOS

**Comment by mrkubax10** [comment]
Ubuntu Touch and probably Mobian **only** works on Android devices because Android uses Linux kernel and that allows to reuse device vendor drivers. iOS on the other hand is a completely different operating system and iPhones are different devices too.

**Comment by Anatoliy6463** [comment]
> Ubuntu Touch and probably Mobian **only** works on Android devices because Android uses Linux kernel and that allows to reuse device vendor drivers. iOS on the other hand is a completely different operating system and iPhones are different devices too.

Ubuntu Touch works on iPhone 7(it's old enough to get support)

**Comment by mrkubax10** [comment]
I don't see it there: https://devices.ubuntu-touch.io/
Unless you mean this: https://iphone.ubports.com/ ;)

**Comment by Anatoliy6463** [comment]
Oh, that was April fools. I hate them.

**Comment by Lulu-Games** [comment]
> > But I get the same problem on a IPAD pro (with defenetly enough ram)
> 
> I think it's because of old iOS

Not possible (IOS 17)

**Comment by Anatoliy6463** [comment]
> > > But I get the same problem on a IPAD pro (with defenetly enough ram)
> > 
> > 
> > I think it's because of old iOS
> 
> Not possible (IOS 17)

iOS 17 is very buggy(why apple released it?)

**Comment by Vankata453** [comment]
> iOS 17 is very buggy(why apple released it?)

Please keep discussions on-topic.

**Comment by Anatoliy6463** [comment]
@Lulu-Games said that his iPad Pro isn't loading wasm and he has iOS 17 which is very bad(and buggy) version of iOS

**Comment by tulpenkiste** [comment]
> @Lulu-Games said that his iPad Pro isn't loading wasm and he has iOS 17 which is very bad(and buggy) version of iOS

This has no relevancy to the issue topic or with WASM. Please refrain from off-topic discussions.

**Comment by Lulu-Games** [comment]
Working WASM would fix the IOS problem, no one needs the executable on the phone if there is the online build

**Comment by Anatoliy6463** [comment]
Try Firefox, it's available in App Store

**Comment by Lulu-Games** [comment]
I tried already (iPhone iPad) and it doesn't work at all
